dashpay/src/main/kotlin/org/dashj/platform/contracts/wallet/TxMetadata.kt

```diff
@@ -0,0 +1,102 @@
+/**
+ * Copyright (c) 2020-present, Dash Core Group
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+package org.dashj.platform.contracts.wallet
+
+import java.util.Date
+import kotlin.collections.HashMap
+import org.bitcoinj.core.ECKey
+import org.dashj.platform.dapiclient.model.DocumentQuery
+import org.dashj.platform.dpp.document.Document
+import org.dashj.platform.dpp.document.DocumentCreateTransition
+import org.dashj.platform.dpp.document.DocumentsBatchTransition
+import org.dashj.platform.dpp.identifier.Identifier
+import org.dashj.platform.dpp.identity.Identity
+import org.dashj.platform.sdk.platform.Platform
+
+class TxMetadata(
+    val platform: Platform
+) {
+
+    companion object {
+        const val DOCUMENT: String = "dashwallet.tx_metadata"
+    }
+
+    fun create(
+        keyIndex: Int,
+        encryptionKeyIndex: Int,
+        encryptedMetadata: ByteArray,
+        identity: Identity,
+        id: Int,
+        signingKey: ECKey
+    ): Document {
+        val profileDocument = createDocument(keyIndex, encryptionKeyIndex, encryptedMetadata, identity)
+        profileDocument.createdAt = Date().time
+
+        val transitionMap = hashMapOf(
+            "create" to listOf(profileDocument)
+        )
+
+        val transition = signAndBroadcast(transitionMap, identity, id, signingKey)
+
+        val rawDocument = transition.transitions[0].toObject().toMutableMap()
+        rawDocument["\$ownerId"] = transition.ownerId
+
+        return platform.dpp.document.createFromObject(rawDocument)
+    }
+
+    private fun signAndBroadcast(
+        transitionMap: HashMap<String, List<Document>>,
+        identity: Identity,
+        id: Int,
+        signingKey: ECKey
+    ): DocumentsBatchTransition {
+        val profileStateTransition =
+            platform.dpp.document.createStateTransition(transitionMap)
+        profileStateTransition.sign(identity.getPublicKeyById(id)!!, signingKey.privateKeyAsHex)
+        platform.broadcastStateTransition(profileStateTransition)
+        return profileStateTransition
+    }
```

Comment:
this function is probably in a few places now.  It could also be modified to take a `ByteArray` as the signingKey.  DPP would convert it back to the correct class before the signing is done.  Not sure if that adds extra processing time.  it depends on if the Public Key is generated when an ECKey object is recreated, which is a slow process.

The reason for possibly specifying the `signingKey` as a `ByteArray` would be to support BLS keys.  The identity indicates what kind of key it would be based on the `id` parameter.